### PR TITLE
CODETOOLS-7903106: Reduce dependence on the "form" of dependencies

### PR DESCRIPTION
**Pull request by jonathan-gibbons**
Please review a medium-sized refactoring of the `jtreg` build and top level of the code.

The underlying goal is to eliminate knowledge from the makefiles and source code of specific transitive dependencies for dependencies like JUnit and TestNG, so that it is easier/simpler to update the (versions of the) dependencies without having to update the makefiles or source code. It becomes sufficient to just update scripts like `build.sh` that download dependencies before invoking the makefiles.   In particular, it isolates the makefiles and code from the choice to use discrete jars or "uber-jars" for the dependencies.

To this end, the makefile variables for a dependency and its dependencies are replace by a single variable that lists them all. For example, `TESTNG_JAR`, `GOOGLE_GUICE_JAR` and `JCOMMANDER_JAR` are combined into a single variable, `TESTNG_JARS`.   Likewise `JUNIT_JAR` and `HAMCREST_JAR` are combined into `JUNIT_JARS`.  Similarly, the makefile variables for the various legal notices are combined into `TESTNG_NOTICES` and `JUNIT_NOTICES`.

Code in `Tool.java` used to have specific knowledge of these dependencies and their transitive dependencies.
To address this, the makefiles now create a new `jar.properties` file, listing one or more jar files for each named dependency.  Within the code, the old `JarFinder` class is replaced by a new `JarManager` class that reads the new `jar.properties` file to determine the set of jar files needed for each dependence.

For anyone building `jtreg` with `build.sh`, the change should be completely transparent.
For anyone running `jtreg`, the change should be completely transparent.

For anyone using custom scripts that invoke the makefiles directly, the set of variables that need to be specified has changed.

As part of this work, the README in the repo has been split into a list of references to other documentation, with the content about building `jtreg` moved and updated to a new file, `doc/building.md`.

A side effect of this work is that version numbers in the names of imported jar files are now preserved, where previously each file was renamed as it was imported, to eliminate the version. This has a (beneficial) ripple effect on the code for `-version` that reports the versions of dependencies; it also has a ripple effect into some tests that previously assumed simple names for these jar files.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [CODETOOLS-7903106](https://bugs.openjdk.java.net/browse/CODETOOLS-7903106): Reduce dependence on the "form" of dependencies

### Reviewers
 * [Iris Clark](https://openjdk.java.net/census#iris) (@irisclark - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jtreg pull/61/head:pull/61` \
`$ git checkout pull/61`

Update a local copy of the PR: \
`$ git checkout pull/61` \
`$ git pull https://git.openjdk.java.net/jtreg pull/61/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 61`

View PR using the GUI difftool: \
`$ git pr show -t 61`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jtreg/pull/61.diff">https://git.openjdk.java.net/jtreg/pull/61.diff</a>

</details>

### CODE REVIEW

File: make/jtreg.gmk [hunk 1]
```diff
@@ -84,7 +87,21 @@ RESOURCES.com.sun.javatest.regtest = \
...
 	$(CLASSDIR)/com/sun/javatest/regtest/tool/i18n.properties \
-	$(CLASSDIR)/com/sun/javatest/regtest/tool/jtlogo.png
+	$(CLASSDIR)/com/sun/javatest/regtest/tool/jtlogo.png \
+	$(CLASSDIR)/com/sun/javatest/regtest/tool/jars.properties
+
+$(BUILDDIR)/classes/com/sun/javatest/regtest/tool/jars.properties: \
+	$(ASMTOOLS_JAR) \
+	$(JCOV_JAR) $(JCOV_NETWORK_SAVER_JAR) \
+	$(TESTNG_JARS) \
+	$(JUNIT_JARS)
```
Comment (irisclark) [inline make/jtreg.gmk:97]: Missing trailing "\\"?

**Review by irisclark — APPROVED** [review]
Great re-factor!

File: make/jtreg.gmk (hunk 1 above)
Comment (jonathan-gibbons) [reply make/jtreg.gmk:97]: Will check and fix as needed

**Pull request closed without merging**